src/App.js

```diff
@@ -1,16 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatMessagesData, setChatMessagesData] = useState(chatMessages);
+  const users = [];
+  let likesCount = 0;
+  let user1 = '';
+  let user2 = '';
+
+  const updateChatData = (updatedMessage) => {
+    const messages = chatMessagesData.map((message) => {
+      if (message.id === updatedMessage.id) {
+        return updatedMessage;
+      } else {
+        return message;
+      }
+    });
+
+    setChatMessagesData(messages);
+  };
+
+  for (let i = 0; i < chatMessagesData.length; i++) {
+    let message = chatMessagesData[i];
+    if (users.indexOf(message['sender']) === -1) {
+      users.push(message['sender'])
+    }
+    if (message['liked']) {
+      likesCount++;
+    }
+  }
```

Comment:
Putting this logic into a method like `calculateHeartCount` and then calling that function within the paragraph tag would help make this file a little more organized and readable. 

```javascript
<p className="header">{calculateHeartCount()} 💙s</p>
```

Having said that, your logic as it is right now would need to be reworked so that when the method is called it would return an integer